api/proto/teleport/legacy/types/types.proto

```diff
@@ -3440,6 +3440,18 @@ message AccessRequestConditions {
     (gogoproto.nullable) = false,
     (gogoproto.jsontag) = "kubernetes_resources,omitempty"
   ];
+
+  // Reason defines settings for the reason for the access provided by the user.
+  AccessRequestConditionsReason Reason = 9 [(gogoproto.jsontag) = "reason,omitempty"];
+}
+
+// AccessRequestConditionsReason defines settings for the reason for the access provided by the
+// user.
+message AccessRequestConditionsReason {
+  // Required indicates that reason is required for all Access Requests requesting allowed roles or
+  // resources searchable with search_as_roles. It applies only to users who have this role
+  // assigned.
+  bool Required = 1 [(gogoproto.jsontag) = "required,omitempty"];
```

Comment:
in the past, we had enormous trouble handling bools, so it's a general recommendation to transform them into enums to avoid situations where we need to add a third state.

Reply:
>In this case the API type is translated to the user YAML object

we use the Marshal/Unmarshal functions for that purpose. We have several examples in types/ package

Reply:
Please have the manifest match the API type and don't play with custom marshallers. Else the terraform and operator generated code differs from the YAML manifest and this is both hard to implement and a terrible UX (users have to deal with changing types depending on how they create the resource)

Reply:
@kopiczko I have no objections to making this a string/enum. I think Tiago has a point - FWIW we always say "it's unlikely this will require a third value" and then there comes a use-case that requires a third value so we have to do deprecations, migrations, etc. :) As a contrived example, what if at some point we decide to introduce a cluster-level setting for required reason or something and have some roles specify whether it's required explicitly and some roles use cluster default, that would require another value, say, "default" or having to deal with unset values, etc. Again, maybe contrived example but feels plausible.

So yeah doing something like

```
spec:
  allow:
    request:
      reason:
        mode: (required|optional)
```

would leave us room for extensibility if needed.

Reply:
Also, if possible please use string and not an actual proto enum.
Teleport Kubernetes Operator and Terraform Provider do not play well with enums and usually require the user to enter the enum's int value.

Eg
<img width="905" alt="image" src="https://github.com/user-attachments/assets/3e2d016e-de2d-4487-bbd8-806411c15793">
https://goteleport.com/docs/reference/terraform-provider/resources/auth_preference/